docs/contributors/guide/cherry-pick/cherry-pick.md

```diff
@@ -0,0 +1,129 @@
+use case for this task is backporting PRs from master to release branches.
+
+- [Prerequisites](#prerequisites)
+- [What Kind of PRs are Good for Cherry Picks](#what-kind-of-prs-are-good-for-cherry-picks)
+- [Initiate a Cherry Pick](#initiate-a-cherry-pick)
+- [Cherry Pick Review](#cherry-pick-review)
+- [Searching for Cherry Picks](#searching-for-cherry-picks)
+- [Troubleshooting Cherry Picks](#troubleshooting-cherry-picks)
+- [Batch Merging Cherry Picks](#batch-merging-cherry-picks)
+
+## Prerequisites
+
+- A pull request merged against the `master` branch.
+- The release branch exists (example: [`release-5.0`](https://github.com/pingcap/tiflow/tree/release-5.0))
+
+## What Kind of PRs are Good for Cherry Picks
+
+The emphasis is on critical bug fixes, e.g.,
+
+- No workaround to the issue
+- Loss of data
+- Memory corruption
+- Panic, crash, hang
+- Security
+
+Some not so critical but important matters, e.g.,
+
+- Tests(new tests and fixes for unstable tests)
+- Metrics And Logging
+- CI Improvements
+
+A bugfix for a functional issue (not a data loss or security issue) does not qualify as a critical bug fix.
+
+If you are proposing a cherry pick and it is not a clear and obvious critical bug fix, please reconsider. If upon
+reflection you wish to continue, bolster your case by supplementing your PR with e.g.,
+
+- A GitHub issue detailing the problem
+
+- Scope of the change
+
+- Risks of adding a change
+
+- Risks of associated regression
+
+- Testing performed, test cases added
+
+## Initiate a Cherry Pick
+
+- Add `needs-cherry-pick-release-{version}` label to the PR, the robot will then automatically create a new PR for the
+  cherry pick.
+- Your cherry pick PR will immediately get the `do-not-merge/cherry-pick-not-approved` label.
+
+## Cherry Pick Review
+
+Unlike other PRs, cherry pick PR will directly copy the LGTMs of the original PR, but we still require `/merge` to
+trigger the merge PR.
+
+The same release note requirements apply as normal pull requests, except the release note stanza will auto-populate from
+the master branch pull request from which the cherry pick originated.
+
+If this is unsuccessful, the `do-not-merge/release-note-label-needed` label will be applied and the cherry pick author
+must edit the pull request description to add a release note or include in a comment the `/release-note-none` command.
+
+Cherry pick pull requests are reviewed slightly differently than normal pull requests on the `master` branch in that
+they:
+
+- Are by default expected to be `type/bug-fix`.
+
+- The original change to the `master` branch is expected to be merged for some time and no related CI failures or test
+  flakiness must be discovered.
+
+- Milestones must be set on the PR reflecting the milestone for the target release branch (for example, milestone v5.3.2
+  for a cherry pick onto branch
+  `release-5.3`). This is normally done for you by manually.
```

Comment:
do we have commands for this operations? it seems that not everyone have edit label privileges

Reply:
what's the purpose of this action? I don't like extra actions 😂 
- we can see the related tags for a commit to know the fix versions https://github.com/pingcap/tiflow/commit/07a7e0e1877cf95f97784df9b8e309e174f6575f
- we should always merge every approved PRs with `type/cherry-pick-for-release-x.y` when preparing release

Reply:
You can see in milestone which PRs were merged this time, branching is not possible, you need to do a comparison to do it.